util/udt_util.cc

```diff
@@ -340,4 +340,33 @@ Status ValidateUserDefinedTimestampsOptions(
   return Status::InvalidArgument(
       "Unsupported user defined timestamps settings change.");
 }
+
+std::tuple<std::optional<Slice>, std::optional<Slice>>
+MaybeAddTimestampsToRange(const Slice* start, const Slice* end, size_t ts_sz,
+                          std::string* start_with_ts, std::string* end_with_ts,
+                          bool exclusive_end) {
+  std::optional<Slice> ret_start =
+      (start == nullptr) ? std::nullopt : std::optional<Slice>(*start);
+  std::optional<Slice> ret_end =
+      (end == nullptr) ? std::nullopt : std::optional<Slice>(*end);
+
+  if (start != nullptr && ts_sz != 0) {
+    // Maximum timestamp means including all keys with any timestamp for start
+    AppendKeyWithMaxTimestamp(start_with_ts, *start, ts_sz);
+    ret_start = Slice(*start_with_ts);
+  }
+  if (end != nullptr && ts_sz != 0) {
+    if (exclusive_end) {
+      // Append a maximum timestamp as the range limit is exclusive:
+      // [start, end)
+      AppendKeyWithMaxTimestamp(end_with_ts, *end, ts_sz);
+    } else {
+      // Append a minimum timestamp to end so the range limit is inclusive:
+      // [start, end]
+      AppendKeyWithMinTimestamp(end_with_ts, *end, ts_sz);
+    }
+    ret_end = Slice(*end_with_ts);
+  }
+  return std::make_tuple(ret_start, ret_end);
```

Comment:
Maybe something like this would flow a bit better; what do you think?

```
std::optional<Slice> ret_start, ret_end;

if (start) {
   if (ts_sz == 0) {
      ret_start = *start;
   } else {
      AppendKeyWithMaxTimestamp(start_with_ts, *start, ts_sz);
      ret_start = *start_with_ts;
   }
} 

// ... populate ret_end similarly

return std::make_tuple(ret_start, ret_end);
```

Reply:
Nice, now if flows! Thank you for the suggestion.